### PR TITLE
rkvm: init at ...

### LINKED ISSUE #108276: package request: rkvm: Virtual KVM switch for Linux machines
**Project description**
It's like synergy or barrier, but supports wayland and windows as well.

I started packaging it [here](https://github.com/colemickens/nixcfg/tree/main/pkgs/rkvm), but it doesn't build yet (`build.rs` fails to find the evdev files it wants).

**Metadata**

* homepage URL: https://github.com/htrefil/rkvm
* source URL: https://github.com/htrefil/rkvm
* license: mit
* platforms: linux

### PR DESCRIPTION
**Pull request by colemickens**
###### Motivation for this change
Fixes #108276 by adding `rkvm`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/rkvm/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    owner = "htrefil";
+    repo = pname;
+    rev = "ec404c69b38f7feff5103f898612734ee8d7ee95";
+    sha256 = "sha256-YZXHbZ71EsSyAtTenQ4rgp1fJbatkk7BW4Cmr/RpyXc=";
+  };
+
+  cargoSha256 = "sha256-1sxP74xt3DGCAhY6Dolaz2JB3Q6CTvtkgln08AMGmc0=";
+
+  postPatch = ''
+    sed -i 's|.clang_arg("-I/usr/include/libevdev-1.0/")|.clang_arg("-I${libevdev}/include/libevdev-1.0").clang_arg("-I${linuxHeaders}/include")|g' ./input/build.rs
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/rkvm/default.nix:18]: Maybe we should do a patch for this because this seems quite complex.

Comment (colemickens) [reply pkgs/applications/misc/rkvm/default.nix:18]: Hi @SuperSandro2000, thanks for looking so fast. I've left this as a draft as someones helping me in IRC. I plan to file bugs upstream, wait for them, something before marking this as ready. Or at least writing this in a slightly more readable way at least.

**Comment by colemickens** [comment]
With the version I just pushed, this is the trailing result of `nix build -L .#rkvm`:

```
rkvm-unstable-ec404c69> error: failed to run custom build command for `input v0.2.0 (/build/source/input)`
rkvm-unstable-ec404c69> Caused by:
rkvm-unstable-ec404c69>   process didn't exit successfully: `/build/source/target/release/build/input-8e4ca7164380d3b7/build-script-build` (exit code: 101)
rkvm-unstable-ec404c69>   --- stdout
rkvm-unstable-ec404c69>   cargo:rerun-if-changed=glue/glue.h
rkvm-unstable-ec404c69>   cargo:rustc-link-lib=evdev
rkvm-unstable-ec404c69>   --- stderr
rkvm-unstable-ec404c69>   /nix/store/0avnch9kw2il7xqssha6ahxx03i38xi9-linux-headers-5.9.8/include/linux/input.h:13:10: fatal error: 'sys/time.h' file not found
rkvm-unstable-ec404c69>   /nix/store/0avnch9kw2il7xqssha6ahxx03i38xi9-linux-headers-5.9.8/include/linux/input.h:13:10: fatal error: 'sys/time.h' file not found, err: true
rkvm-unstable-ec404c69>   thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()', input/build.rs:21:10
rkvm-unstable-ec404c69>   note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
rkvm-unstable-ec404c69> warning: build failed, waiting for other jobs to finish...
rkvm-unstable-ec404c69> error: build failed
```

**Comment by colemickens** [comment]
TODO: the binaries need to be prefixed with `rkvm-`. These don't make good random binary names in PATH: `client`, `server`, `certificate-gen`.

**Comment by colemickens** [comment]
not sure if it's working anyway: https://github.com/htrefil/rkvm/issues/7

**Comment by colemickens** [comment]
I'm abandoning this. I'm happy doing everything from a single machine nowadays. If someone else wants to pick, please see @peterhoeg's comment, since I'll probably prune this branch soon.

**Pull request closed without merging**